Sources/Controllers/2.0/OAEditFavoriteViewController.mm

```diff
@@ -0,0 +1,1024 @@
+                                            [self deleteFavoriteItem:self.favorite];
+                                            [self dismissViewControllerAnimated:YES completion:nil];
+                                        }],
+      nil] show];
+}
+
+- (void) deleteFavoriteItem:(OAFavoriteItem *)favoriteItem
+{
+    [OAFavoritesHelper deleteFavoriteGroups:nil andFavoritesItems:@[favoriteItem]];
+}
+
+- (void) removeExistingItemFromCollection
+{
+    NSString *favoriteTitle = [self.favorite getFavoriteName];
+    for(const auto& localFavorite : [OsmAndApp instance].favoritesCollection->getFavoriteLocations())
+    {
+        if ((localFavorite != self.favorite.favorite) &&
+            [favoriteTitle isEqualToString:localFavorite->getTitle().toNSString()])
+        {
+            [OsmAndApp instance].favoritesCollection->removeFavoriteLocation(localFavorite);
+            break;
+        }
+    }
+}
+
+- (void) removeNewItemFromCollection
+{
+    _app.favoritesCollection->removeFavoriteLocation(self.favorite.favorite);
+    [_app saveFavoritesToPermamentStorage];
+}
+
+-(void) clearButtonPressed:(UIButton *)sender
+{
+    NSIndexPath *indexPath = [self indexPathForCellContainingView:sender inTableView:self.tableView];
+    OATextInputFloatingCellWithIcon *cell = [self.tableView cellForRowAtIndexPath:indexPath];
+    [self.tableView beginUpdates];
+    
+    cell.textField.text = @"";
+    
+    NSDictionary *item = _data[indexPath.section][indexPath.row];
+    NSString *key = item[@"key"];
+    if ([key isEqualToString:kNameKey])
+        self.name = @"";
+    else if ([key isEqualToString:kDescKey])
+        self.desc = @"";
+    else if ([key isEqualToString:kAddressKey])
+        self.address = @"";
+
+    cell.fieldLabel.hidden = YES;
+    cell.textFieldTopConstraint.constant = 0;
+    cell.textFieldBottomConstraint.constant = 0;
+    
+    [self generateData];
+    [self.tableView endUpdates];
+}
+
+- (NSIndexPath *)indexPathForCellContainingView:(UIView *)view inTableView:(UITableView *)tableView
+{
+    CGPoint viewCenterRelativeToTableview = [tableView convertPoint:CGPointMake(CGRectGetMidX(view.bounds), CGRectGetMidY(view.bounds)) fromView:view];
+    NSIndexPath *cellIndexPath = [tableView indexPathForRowAtPoint:viewCenterRelativeToTableview];
+    return cellIndexPath;
+}
+
+#pragma mark - UITableViewDataSource
+
+- (NSInteger) numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return _data.count;
+}
+
+- (NSInteger) tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    return _data[section].count;
+}
+
+- (nonnull UITableViewCell *) tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath
+{
+    NSDictionary *item = _data[indexPath.section][indexPath.row];
+    NSString *cellType = item[@"type"];
+    
+    if ([cellType isEqualToString:kTextInputFloatingCellWithIcon])
+    {
+        OATextInputFloatingCellWithIcon *resultCell = nil;
```

Comment:
Sometimes the input textfield is deleted while rotating the device. Please double-check and fix it